torch/distributed/elastic/rendezvous/dynamic_rendezvous.py

```diff
@@ -1231,13 +1235,18 @@ def create_handler(
             _get_timeout(params, "last_call"),
             _get_timeout(params, "close"),
         )
+        hostname = ''
+        param_hostname = _try_parse_host_from_endpoint(params.endpoint)
```

Comment:
This endpoint could be a different host right?

like if you rendezvous with endpoint == "rank0" on rank 1 machine we wouldn't want to grab rank0 as the current address? am I misunderstanding this?